src/box/box.cc

```diff
@@ -3615,6 +3615,38 @@ on_wal_checkpoint_threshold(void)
 	gc_trigger_checkpoint();
 }
 
+static const uint32_t timeout_check_period = 1000;
+static uint32_t period = 0;
```

Comment:
Why do you need them to be global? Can you put them in `box_check_deadline`?

Also, we usually define int constants in enums, not as variables:

```C
enum { my_constant = 123 };
```

Reply:
Put in `box_deadline_is_up`.